scipy/linalg/tests/test_lapack.py

```diff
@@ -1705,3 +1705,201 @@ def test_getc2_gesc2():
+    assert_allclose(x, x_soln, rtol=rtol, atol=atol)
+
+    U = np.diag(df, 0) + np.diag(duf, 1) + np.diag(du2f, 2)
+    L = np.eye(n, dtype=dtype)
+
+    for i, m in enumerate(dlf):
+        # L is given in a factored form.
+        # See http://www.hpcavf.uclan.ac.uk/softwaredoc/sgi_scsl_html/sgi_html/ch03.html
+        piv = ipiv[i] - 1
+        # right multiply by permutation matrix
+        L[:, [i, piv]] = L[:, [piv, i]]
+        # right multiply by Li, rank-one modification of identity
+        L[:, i] += L[:, i+1]*m
+
+    # one last permutation
+    i, piv = -1, ipiv[-1] - 1
+    # right multiply by final permutation matrix
+    L[:, [i, piv]] = L[:, [piv, i]]
+
+    # check that the outputs define an LU decomposition of A
+    assert_allclose(A, L @ U, atol=atol, rtol=rtol)
+
+    # assert that the outputs are of correct type or shape
+    # rcond should be a scalar
+
+    assert_(hasattr(rcond, "__len__") is not True,
+            "rcond should be scalar but is {}".format(rcond))
+    # ferr should be length of # of cols in x
+    assert_(ferr.shape[0] == b.shape[1], "ferr.shape is {} but shoud be {},"
+            .format(ferr.shape[0], b.shape[1]))
+    # berr should be length of # of cols in x
+    assert_(berr.shape[0] == b.shape[1], "berr.shape is {} but shoud be {},"
+            .format(berr.shape[0], b.shape[1]))
+
+    # test with singular matrix
+    # no need to test with fact "F" since ?gttrf already tests for these.
+    if fact == "N":
+        d[n-1] = 0
+        dl[n-2] = 0
+
+        # solve using routine
+        gtsvx_out = gtsvx(dl, d, du, b)
+        dlf, df, duf, du2f, ipiv, x_soln, rcond, ferr, berr, info = gtsvx_out
+        # test for the singular matrix.
+        assert_(d[info - 1] == 0,
+                "?gtsvx: d[info-1] is {}, not the illegal value"
+                .format(d[info - 1]))
+
+    else:
+        # assuming that someone is using a singular factorization
+        df[n-1] = 0
+        duf[n-2] = 0
+
+        gtsvx_out = gtsvx(dl, d, du, b, fact=fact, dlf=dlf_, df=df_, duf=duf_,
+                          du2=du2f_, ipiv=ipiv_)
+        dlf, df, duf, du2f, ipiv, x_soln, rcond, ferr, berr, info = gtsvx_out
+        # info should not be zero and should provide index of illegal value
+        assert_(df[info - 1] == 0,
+                "?gtsvx: df[info-1] is {}, not the illegal value"
+                .format(d[info - 1]))
+
+    if dtype in REAL_DTYPES:
+        trans = "T"
+    else:
+        trans = "C"
+    A_trans = A.conj().T
+    # Test with transposed matricies
+    # restore these
+    du = inputs_cpy[2]
+    d = inputs_cpy[1]
+    dl = inputs_cpy[0]
+
+    b = A_trans@x
+    gtsvx_out = gtsvx(dl, d, du, b, trans=trans)
+    dlf, df, duf, du2f, ipiv, x_soln, rcond, ferr, berr, info = gtsvx_out
+    # test that x_soln matches the expected x
+    assert_allclose(x, x_soln, rtol=rtol, atol=atol)
+
+    U = np.diag(df, 0) + np.diag(duf, 1) + np.diag(du2f, 2)
```

Comment:
Since we're checking that the factors represent an LU decomposition in `gttrf`, too, please move all this into a function `L, U = gttrf2lu(dlf, df, duf, du2f, ipiv)` (or similar). Use the function above (and here if you can't parameterize transposal option`, and after the `gttrf` is merged, we should modify its test to use the function.